php/Blockonomics.php

```diff
@@ -487,16 +487,13 @@ public function create_new_order($order_id, $crypto){
             exit(json_encode(array("error"=>$responseObj->response_message)));
         }
         $address = $responseObj->address;
```

Comment:
Instead of changing rest of the code, I prefer we check if the address is already existing here and exit with error in the same way we are existing in line 487. 

_Duplicate Address Error.  This is Temporary error, Please try again_

Reply:
If both addresses are returned close to the same time, the 1st matching address may not be saved to the database yet. 

It is best to let the database handle checking for duplicates when saving the new address, which is currently performed once the complete order is available:
https://github.com/blockonomics/woocommerce-plugin/pull/263/files#diff-a3307e680eb76b9d93556c9ae2e5dc134b3526246480730fb8ddf49186da29ebR561

Another option to reduce the code impact is to update the database to use KEY orderkey (order_id, address, crypto), as currently, the 2nd order id is replacing the 1st in the database due to using `replace` to insert or replace:
https://github.com/blockonomics/woocommerce-plugin/blob/master/blockonomics-woocommerce.php#L485

Reply:
>If both addresses are returned close to the same time, the 1st matching address may not be saved to the database yet.
I agree with this

I am unsure how this will give [an error](https://github.com/blockonomics/woocommerce-plugin/pull/263/files#diff-a3307e680eb76b9d93556c9ae2e5dc134b3526246480730fb8ddf49186da29ebR561) when an existing addresses is inserted. The basic problem is we are lacking UNIQUE tag on address column

A select to check uniqueness immediately followed by wp update/replace should work in 99.9% of cases, but it still open to some [race conditions](https://stackoverflow.com/questions/48475909/if-record-exists-when-using-wpdb-insert-dont-insert-and-dont-throw-failure) 

The better solution is to create a new address column having UNIQUE tag and use this column from now.  We cannot put UNIQUE on existing column as it may already have non uniques. Then if _wp->replace_ gives exception we handle it and then show the error

Reply:
Currently, there may be two orders with the same address in the post meta, however, the database will not have any duplicate addresses as there is already a unique key on the address:
https://github.com/blockonomics/woocommerce-plugin/blob/master/blockonomics-woocommerce.php#L484 ​
This leads to the order_id being replaced by the second order with the same address as we are using replace to insert or replace:
https://github.com/blockonomics/woocommerce-plugin/blob/master/php/Blockonomics.php#L553

[This will give an error](https://github.com/blockonomics/woocommerce-plugin/pull/263/files#diff-a3307e680eb76b9d93556c9ae2e5dc134b3526246480730fb8ddf49186da29ebR561) when an existing address is inserted because the insert already checks for a duplicate key.

In order to get the existing `replace` to work instead of using the insert, a unique key should be placed on order_id, address, and crypto, as we currently only use the address and crypto:
https://github.com/blockonomics/woocommerce-plugin/blob/master/blockonomics-woocommerce.php#L485

Reply:
Ok understood. Apologies. Wrong understanding by me. 

It is actually well written code
Please add code comments to explain workflow/motivation as later it will be difficult to understnad